alf/metrics/metrics.py

```diff
@@ -34,12 +34,12 @@ def __init__(self, max_len, dtype):
         super().__init__()
         shape = (max_len, )
         self.dtype = dtype
-        self._max_len = torch.nn.Parameter(
-            torch.tensor(max_len, dtype=torch.int64), requires_grad=False)
-        self._buffer = torch.nn.Parameter(
-            torch.zeros(shape, dtype=dtype), requires_grad=False)
-        self._head = torch.nn.Parameter(
-            torch.zeros((), dtype=torch.int64), requires_grad=False)
+        self._max_len = torch.tensor(max_len, dtype=torch.int64)
+        self.register_buffer(tensor=self._max_len, name='max_len')
```

Comment:
```python
self.register_buffer('_max_len', torch.tensor(max_len, dtype=torch.int64))
```
otherwise self.max_len is still valid because register_buffer will add 'max_len' as attribute of self

Reply:
Ah, only briefly read the API..